packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -0,0 +1,257 @@
+
+    constructor() {
+        super();
+
+        this.handleItemsChange();
+    }
+
+    private getParentTreeNode() {
+        const parentNode: Element | null | undefined = this.parentElement!.closest(
+            "[role='tree']"
+        );
+        return parentNode;
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        const parentTreeNode: any = this.getParentTreeNode();
+        if (parentTreeNode) {
+            this.notifier = Observable.getNotifier(parentTreeNode);
+            this.notifier.subscribe(this, "render-collapsed-nodes");
+            this.renderCollapsedChildren =
+                parentTreeNode.getAttribute("render-collapsed-nodes") === "true";
+        }
+    }
+
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+        if (this.notifier) {
+            this.notifier.unsubscribe(this, "render-collapsed-nodes");
+        }
+    }
+
+    public handleChange(source: any, propertyName: string) {
+        switch (propertyName) {
+            case "render-collapsed-nodes":
+                this.renderCollapsedChildren =
+                    source.getAttribute("render-collapsed-nodes") === "true";
+                break;
+        }
+    }
+
+    public handleFocus = (e: Event): void => {
+        if (e.target === e.currentTarget) {
+            this.focusable = true;
+        }
+    };
+
+    public handleBlur = (e: FocusEvent): void => {
+        if (e.target !== e.currentTarget) {
+            return;
+        }
+
+        this.focusable = false;
+    };
+
+    public handleKeyDown = (e: KeyboardEvent): void | boolean => {
+        if (e.target !== e.currentTarget) {
+            return;
+        }
+
+        switch (e.keyCode) {
+            case keyCodeArrowLeft:
+                this.handleArrowLeft();
+                break;
+            case keyCodeArrowRight:
+                this.handleArrowRight();
+                break;
+            case keyCodeArrowDown:
+                // preventDefault to ensure we don't scroll the page
+                e.preventDefault();
+                this.focusNextNode(1);
+                break;
+            case keyCodeArrowUp:
+                // preventDefault to ensure we don't scroll the page
+                e.preventDefault();
+                this.focusNextNode(-1);
+                break;
+            case keyCodeEnter:
+                this.handleSelected(e);
+                break;
+            case keyCodeSpace:
+                this.handleSpaceBar();
+                break;
+        }
+
+        return true;
+    };
+
+    public handleExpandCollapseButtonClick = (): void => {
+        if (!this.disabled) {
+            this.setExpanded(!this.expanded);
+        }
+    };
+
+    public handleContainerClick = (e: MouseEvent): void => {
+        const expandButton: HTMLElement | null = this.expandCollapseButton;
+        if (
+            (!isHTMLElement(expandButton) ||
```

Comment:
You check `isHTMLElement` for the same value twice - maybe store the result?

Reply:
will do

Reply:
done and done